src/utils/storage.ts

```diff
@@ -65,6 +70,51 @@ export const setStorageItem = async <Value extends unknown>({
 
   // Dispatch our internal event
   window.dispatchEvent(new Event(`amino:storage-${type}`));
+};
+
+export const setStorageItemWithLifetime = async <Value extends unknown>({
+  key,
+  lifetime,
+  type,
+  value,
+}: SetParams<Value> & { lifetime: Dayjs }) => {
+  const storage =
+    type === 'session' ? window.sessionStorage : window.localStorage;
+
+  const valueToSet = typeof value === 'string' ? value : JSON.stringify(value);
+
+  const valueIsTruthy = !!value;
```

Comment:
My only concern is maybe making this function a parameter, as this truthy logic may not always be desired.